### DISCUSSION TITLE
Extraction of virial data from CP2K Trajectory using "cp2k/aimd_output"

### DISCUSSION THREAD

**Discussion by PWFYeung** [deepmodeling/dpdata#319] [Q&A]
Hi, I am new to dpdata and I have been trying to use dpdata to convert my CP2K data to DeePMD data. I am using a single frame calculation for testing purposes for now. I tried to use `fmt="cp2k/aimd_output"` to convert the CP2K data to deepmd raw files. All the forces, energies are extracted into `.raw` files except for `virial.raw`. However, when I use `fmt="cp2k/output"`, the `virial.raw` can be extracted alongside the other `.raw` files. The problem is "cp2k/output" can only be used for a single frame, but I wish to use dpdata to convert multiframe trajectories.

Does anyone know how to extract the virial data using "cp2k/aimd_output"? Is it supported in dpdata? 

(The cp2k files are attached if you wish to test with it)
[dpdata.zip](https://github.com/deepmodeling/dpdata/files/8998067/dpdata.zip)

**Comment by wanghan-iapcm** [comment]
@robinzyb Could you please take a look? Thanks

**Comment by HuangJiameng** [comment]
https://github.com/deepmodeling/dpdata/pull/324
please follow this PR, it can solve your problem.
